pkg/cmd/experimental/syncgroups/groupdetector/groupdetector.go

```diff
@@ -26,6 +26,10 @@ func (l *GroupBasedDetector) Exists(ldapGroupUID string) (bool, error) {
 		return false, err
 	}
 
+	if group == nil {
+		return false, nil
```

Comment:
I think there may be larger problems. In either case, `QueryForUniqueEntry`, when it retrieves nothing, it should throw an `ErrorEntryNotFound`. Which the detector should recognize as meaning the thing it was looking for doesn't exist.